### PR TITLE
ci: check for unnecessary packages during CI

### PR DESCRIPTION
**Pull request by fallion**
Adds step to Makefile `ci` step that diffs go.mod to help prevent garbage dependencies from getting in the commit.

Example where this could be useful: https://github.com/goreleaser/goreleaser/commit/c338cfb9a1db9922c40fa0149b7ae5c5ad0eabab

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,8 +36,19 @@ lint:
...
 
+# Clean go.mod
+go-mod-tidy:
+	$(eval TMPDIR=$(shell mktemp -d))
+	cp -f go.mod $(TMPDIR)
+	cp -f go.sum $(TMPDIR)
+	go mod tidy -v
+	diff -u $(TMPDIR)/go.mod go.mod
+	diff -u $(TMPDIR)/go.sum go.sum
+	rm -rf $(TMPDIR)
```
Comment (caarlos0) [inline Makefile]: I think we could simplify to `go mod tidy` and checking it with `git status` 🤔

**Comment by fallion** [comment]
I will adjust it.

**Comment by fallion** [comment]
Should be fixed now.

**Comment by caarlos0** [comment]
awesome, thanks!

**Pull request merged**